block/submit.go

```diff
@@ -314,11 +311,36 @@ func (m *Manager) UpdateLastSubmittedHeight(event pubsub.Message) {
 	}
 }
 
+func (m *Manager) GetLastBlockInSettlementTime() (time.Time, error) {
+	lastBlockInSettlement, err := m.Store.LoadBlock(m.LastSettlementHeight.Load())
```

Comment:
I think the LastSettlementHeight may get pruned no?

```go
pruningHeight = min(m.NextHeightToSubmit(), uint64(retainHeight))
```

if that's the case and we fallback to first block it may cause unnecessary halt.

Reply:
true. good catch. i keep last submitted block without pruning to enable skew time calculation and keep code simplicity.